QHyper/problems/community_detection.py

```diff
@@ -9,33 +9,46 @@
 import networkx as nx
 import numpy as np
 import sympy
-from QHyper.util import Expression
 from QHyper.problems.base import Problem
-from QHyper.util import VARIABLES
+from QHyper.constraint import Constraint
+from QHyper.polynomial import Polynomial
 from sympy.core.expr import Expr
+from QHyper.parser import from_sympy
 
 
 @dataclass
 class Network:
     graph: nx.Graph
     resolution: float = field(default=1)
-    modularity_matrix: np.ndarray = field(init=False)
+    full_modularity_matrix: np.ndarray = field(init=False)
+    generalized_modularity_matrix: np.ndarray = field(init=False)
     weight: str | None = field(default=None)
+    community: list | None = field(default=None)
```

Comment:
You can also change line 22 to `resolution: float = 1.0`. In those basic type there is no need to use `field`.
```suggestion
    weight: str | None = None
    community: list | None = None
    full_modularity_matrix: np.ndarray = field(init=False)
    generalized_modularity_matrix: np.ndarray = field(init=False)
```